docs/src/index.md

```diff
@@ -268,6 +268,222 @@ df2 = @eachrow df begin
 end
 ```
 
+## Row-wise transformations with `@byrow`
+
+DataFrames.jl provides the function-wrapper `ByRow`. `ByRow(f)(x, y)`
+is roughly equivalent to `f.(x, y)`, with a few exceptions discussed below. 
```

Comment:
do you discuss these exceptions indeed? The exceptions are:
* `AsTable` case - which I think we do not cover yet
* potentially another type of container can be allocated as we use `map` which is not guaranteed to be 100% consistent with broadcasting in corner cases

(I am not sure it is worth to go into such details in the documentation but I felt it is good to leave this comment)

Reply:
I added a short note about the broadcasting issue. 

I will avoid discussing `AsTable` until we implement that.